### PR TITLE
Style Unification

### PR DESCRIPTION
**Pull request by jeremylt**
This includes style updates to all C code except

* PETSc examples (#726)
* Fluids example (@LeilaGhaffari  feel free to push to this branch, or make a separate one)
* CUDA, OCCA, HIP, and MAGMA backends (@YohannDudouit, @dmed256, @nbeams backend maintainers get to pick the style that is most ideomatic for their backends, feel free to add changes to this PR if you want to change the style in your backends as part of this effort)

I will probably do a check for funny line breaks later, but feel free to comment on any of the renaming.

This PR includes **zero** functionality change - style changes only!

### CODE REVIEW

File: doc/sphinx/source/libCEEDdev.rst [hunk 1]
```diff
@@ -1,6 +1,51 @@
...
+Header Files
+----------------------------------------
+
+Header inclusion for source files should follow the principal of 'include what you use' rather than relying upon transitive ``#include`` to define all symbols.
+
+Every symbol that is used in the source file ``foo.c`` should be defined in ``foo.c``, ``foo.h``, or in a header file ``#include``d in one of these two locations.
+Please check your code by running the tool ``include-what-you-use`` to see recommendations for changes to your source.
+Most issues reported by ``include-what-you-use`` should be fixed; however this rule is flexible to account for differences in header file organization in external libraries.
+
+Header files should be listed in alphabetical order, with installed headers preceding local headers and ``ceed`` headers being listed first.
```
Comment (jedbrown) [inline doc/sphinx/source/libCEEDdev.rst:47]: Does "installed headers" mean those with `<>` while "local" means `""`? Maybe an example?
```c
#include <ceed/ceed.h>
#include <ceed/backend.h>
#include <stdbool.h>
#include <string.h>
#include "ceed-avx.h"
```

**Review by jedbrown — APPROVED** [review]
I'm fine with this whenever you're ready.

File: doc/sphinx/source/libCEEDdev.rst [hunk 2]
```diff
@@ -1,6 +1,51 @@
...
+
+
+Header Files
+----------------------------------------
+
+Header inclusion for source files should follow the principal of 'include what you use' rather than relying upon transitive ``#include`` to define all symbols.
+
+Every symbol that is used in the source file ``foo.c`` should be defined in ``foo.c``, ``foo.h``, or in a header file ``#include``d in one of these two locations.
+Please check your code by running the tool ``include-what-you-use`` to see recommendations for changes to your source.
+Most issues reported by ``include-what-you-use`` should be fixed; however this rule is flexible to account for differences in header file organization in external libraries.
```
Comment (jedbrown) [inline doc/sphinx/source/libCEEDdev.rst:45]: Should we automate this, at least with a build target? Some editors can apply a similar policy, and maybe we could point people at it.

File: doc/sphinx/source/libCEEDdev.rst (hunk 1 above)
Comment (jeremylt) [reply doc/sphinx/source/libCEEDdev.rst:47]: added an example

File: doc/sphinx/source/libCEEDdev.rst (hunk 2 above)
Comment (jeremylt) [reply doc/sphinx/source/libCEEDdev.rst:45]: I got a lot of false hits with `include-what-you-use` that I had to manually adjudicate. I agree that automating this as much as possible is a good idea though.

**Comment by jeremylt** [comment]
I fixed some of the worst line break changes. I'll merge once CI does it's thing and then we can add other style updates in follow-up PRs as desired.

**Pull request merged**